org.alloytools.api/src/main/java/org/alloytools/alloy/dto/SolutionDTO.java

```diff
@@ -0,0 +1,98 @@
+package org.alloytools.alloy.dto;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A Data Transfer Object (DTO) representing a specific Alloy solution.
+ *
+ * This class provides a structured way to encapsulate and transfer data related
+ * to a particular solution generated from an Alloy model.
+ */
+public class SolutionDTO {
+
+    /**
+     * Indicates whether the solution is satisfiable, i.e., whether a valid instance
+     * of the model exists that satisfies all constraints.
+     */
+    public boolean               satisfiable;
+
+
+    /**
+     * A list of InstanceDTO objects that represent the instances associated with
+     * this solution. Each InstanceDTO object within this list encapsulates
+     * information about a specific instance within the solution.
+     */
+    public List<InstanceDTO>     instances = new ArrayList<>();
+
+    /**
+     * The index of the instance that is the final loopback. All traces run to the
+     * last instance and then continue to the loop state. If the solution is not
+     * dynamic or not satisfied, the loop state is -1.
+     */
+    public int                   loopstate = -1;
+
+    /**
+     * The name of the solver
+     */
+    public String                solver;
+
+    /**
+     * The time, in milliseconds since the Unix epoch, at which the solution was
+     * generated.
+     */
+    public long                  utctime;
+
+    /**
+     * The command or operation that generated this solution.
+     */
+    public String                command;
+
+
+    /**
+     * Indicates whether the solution was obtained incrementally. Incremental
+     * solutions can be useful for solving complex problems piece by piece.
+     */
+    public boolean               incremental;
+
+    public String                localtime;
```

Comment:
seems redundant given `utctime` and unnecessary

Reply:
For readability. The utctime is by far the preferred way to use this but people sometimes inspect the json by hand and then utctime is unreadable. Cost is negligible so that is why I included both. Found that some customers really appreciate it.